cmd/dagger/functions.go

```diff
@@ -198,6 +204,11 @@ type FuncCommand struct {
 	// AfterResponse is called when the query has completed and returned a result.
 	AfterResponse func(*FuncCommand, *cobra.Command, *modTypeDef, any) error
 
+	// Whether to load core API types along side the user module types.
+	// TODO: this should become the default behavior once the CLI simplification
+	// work is done, can be removed at that time.
+	IncludeCore bool
```

Comment:
Potential discussion point: would we ever want to allow including dependency types (and not just core)?

Maybe this is not as useful for end-users, but for devs, I can see this being super useful - I have all my dependencies listed in `dagger.json`, I want to play around with those on the CLI, as I write my code.

Or, do interfaces make this kind of irrelevant, since we won't allow accepting/returning foreign concrete types directly?

Reply:
I could see that being potentially useful and provided it's only *direct* dependencies then it would be totally logically consistent since you'd essentially be operating on the same schema that the module itself has (modules have access to both their own APIs and the APIs of their direct deps).

Honestly the only real impediment is that the CLI currently only lets you load one user module at a time (https://github.com/dagger/dagger/issues/5866) and has some assumptions around that. E.g. because there's only one user module loaded you don't have to specify the "main module object" as a top-level command and can instead just start calling functions on it right away (or provide top-level flags for the constructor).
* We can resolve that by just requiring that if multiple user modules are loaded then you have to specify a top-level object/constructor.

But that's all getting out of scope of this PR obviously 😄